### ISSUE TITLE
Details Block pointer is applied to summary element

### ISSUE THREAD

**Issue by jhmonroe** [WordPress/gutenberg#64116]
### Description

The details block uses cursor: pointer on the nested summary element to make the pointer appear so people know it is "linked" to perform an action.

If you add padding or a background to the details block (to create a cards/lozenge look for the detail item)....
You/a user would expect the entire "card" to be the clickable area, as is the case in many UIs)

However, the pointer only appears when hovering over the summary element.

Presumably this was done so the appearance of being "linked" wouldn't appear for the entire details content, but only for the "teaser" text.

But perhaps it would be preferable to move this pointer code to cover the entire details block? Knowing that many people will use details block in card-style designs to emulate modern interface looks.

@richtabor I noticed you had some open issues re: details block, so this might be something you've come across as well as a card-saavy designer.

![Screen Shot 2024-07-30 at 1 42 17 PM](https://github.com/user-attachments/assets/d840b1db-e4a3-4fd6-8768-c06351362a85)

### Step-by-step reproduction instructions

1. Add a background color and padding to the details block to create a "cards" design
2. Hover over the card text and padding on the frontend to view where/when pointer appears and when it does not

### Environment info

WP 6.6

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by carolinan** [comment]
Hi, having a pointer cursor on an element that is not interactive would be an accessibility problem as users may perceive it as clickable when it is not.

Adding the padding on the details does not increase the clickable area.
I think the size of the clickable area is the real issue there though, not the cursor?

**Comment by jhmonroe** [comment]
> Hi, having a pointer cursor on an element that is not interactive would be an accessibility problem as users may perceive it as clickable when it is not.
> 
> Adding the padding on the details does not increase the clickable area. I think the size of the clickable area is the real issue there though, not the cursor?

Yes—I totally agree with you in the use cases where there is no background color! So yes applying it to the entire Details block wouldn't work in those cases. This design with background color is more an example of a common situation people might run into if their site relies heavily on card design style (I'm trying to build only with Core blocks as much as possible).

Since Summary element is coded but can't be controlled in the block editor, the clickable area would need be coded in Core to extend if padding is added to the parent, right?

The clickable area issue in my example (with a background behind details block) kinda reminds me of the old days when someone might accidentally link just the text in a button and not the entire button shape around them. So then it became tedious for users to find the clickable area JUST on the text itself.

At this point in web design and mobile/app design, people just expect an entire shape to be clickable (in it's minimized state).

**Comment by renishsurani** [comment]
Hi @jhmonroe @carolinan,

I have created PR for the #64116. Could you please review this?

PR Link - https://github.com/WordPress/gutenberg/pull/64142

**Comment by jhmonroe** [comment]
Commented on #64142 but adding here in case anyone stumbles on it and wants to add to the discussion

It's interesting to see and inspect how Mozilla handles their Details block and applying padding around Summary as well as separately to the bottom of the .extra div(inside the green box on this page):
https://developer.mozilla.org/en-US/docs/Web/CSS/--*
